src/debug/ee/debugger.cpp

```diff
@@ -16558,6 +16563,154 @@ void Debugger::ReleaseDebuggerDataLock(Debugger *pDebugger)
 #endif // DACCESS_COMPILE
 
 /* ------------------------------------------------------------------------ *
+ * Functions for DebuggerHeap executable memory allocations
+ * ------------------------------------------------------------------------ */
+
+DebuggerHeapExecutableMemoryAllocator::~DebuggerHeapExecutableMemoryAllocator()
+{
+    DebuggerHeapExecutableMemoryPage *currPage = pages;
+    while (currPage != NULL)
```

Comment:
The code in this loop does not look right: it skips the first page in the list and then it can access memory that was freed by VirtualFree.